File: factual.js
//imports
const fs = require('fs');
const axios = require('axios');
const https=require('https');
const bodyParse=require('body-parser');
const express =require("express");
var text; 
const app=express();


var claim="Trump lost the 2020 election";
var key=key;


var url="https://factchecktools.googleapis.com/v1alpha1/claims:search?languageCode=en-us&maxAgeDays=100&pageSize=5&pageToken=1&query="+claim+"&key="+key;

https.get(url,function (res){   
	//console.log(res.statusCode);
	var chunks=[];
	//grab data
	res.on("data",function(data){
		chunks.push(data);
	})
	
	
	res.on("end",function(){
		var jdata=Buffer.concat(chunks);
		var schema=JSON.parse(jdata);

		console.log(schema);
		console.log("\n");
		//gathers info of claim
		var claims=schema.claims;
		

		for(let i=0;i<claims.length;i++){
		
			//Grabs one claim
			var review=claims[i].claimReview;
			var claimant=claims[i].claimant;
			//var claimant=schema[i].claimant;
			console.log(review);
			review=review[0];
	
			//console.log("name: "+review.publisher.name);
			console.log("Claimant: "+claimant);
			console.log("site: "+review.publisher.site);
			console.log("url: "+review.url);
			console.log("title: "+review.title);
			console.log("Rating: "+review.textualRating);
		}
	})
	
})

File: testing/factual.py
import requests
#textlob

myfile=open(r"C:\Users\danie\OneDrive\Desktop\UT_Dallas\ACM_Projects\key.txt","r")
key=myfile.read().replace("\n", "")

claim="trump lost 2020"

url=f"https://factchecktools.googleapis.com/v1alpha1/claims:search?languageCode=en-us&maxAgeDays=100&pageSize=5&pageToken=1&query={claim}&key={key}";
try:
    res = requests.get(url)

    if (res.status_code==200):
        data=res.json()
        if (len(data)!=0):
            
            myclaims=data['claims']
            for i in range(len(myclaims)):
                review=myclaims[i]['claimReview'][0]
                #text,claimant,title,arturl,name,site,rating
                try:
                    text=myclaims[i]['text']
                except:
                    text=None
                try:
                    claimant=myclaims[i]['claimant']
                except:
                    claimant=None
                try:
                    title=review['title']
                except:
                    title=None
                try:
                   newurl=review['url']
                except:
                    newurl=None
                try:
                    name=review['publisher']['name']
                except:
                    name=None
                try:
                    site=review['publisher']['site']
                except:
                    site=None
                try:
                    rating=review['textualRating']
                except:
                    rating=None
                    

                print(f"Origin #{i}")
                print(f"     Claim: {text}")
                print(f"     Source: {claimant}")
                print("Review")
                print(f"     Artical: {title}")
                print(f"     Link: {newurl}")
                print(f"     Publisher: {name} From: {site}")
                print(f"     Claim rated as: {rating}")
                if (i!=len(myclaims)-1):
                    print("________________________________________________")
        else:
            print("No results")
        
    else:
        print("fail")
        
    
except ConnectionError:
    print("Connection Error")